src/main/java/com/github/bpiatek/bbghbackend/ninetyminutes/domain/NinetyMinutesCommentsExtractor.java

```diff
@@ -24,111 +26,46 @@
 
   private final TextToLocalDateTimeParser localDateTimeParser;
 
+  private static final Pattern PATTERN = Pattern.compile("<a class=\"main\">(?<author>[^<]*)</a> - (?<date>.*?) - (?<authorHost>.*?)<br>\n(?<content>.*?)\n<p", Pattern.DOTALL);
+  private static final int PATTERN_GROUPS_COUNT = 4;
+
   NinetyMinutesCommentsExtractor(TextToLocalDateTimeParser localDateTimeParser) {
     this.localDateTimeParser = localDateTimeParser;
   }
 
   @Override
   public List<Comment> getComments(String html) {
+    Matcher matcher = PATTERN.matcher(html);
     List<Comment> comments = new ArrayList<>();
-    final Elements aElements = getPageElements(html, "a");
 
-    for (Element element : aElements) {
-      if (isComment(element)) {
-        String author = getAuthor(element);
-        Comment comment = extractSingleComment(comments, html, author);
-        if(isDuplicated(comment)) {
-          continue;
+    while (matcher.find()) {
+      if (matcher.groupCount() == PATTERN_GROUPS_COUNT) {
+
+        Comment comment = Comment.builder()
+            .author(matcher.group("author"))
+            .content(matcher.group("content"))
+            .dateAdded(localDateTimeParser.parse(matcher.group("date")))
+            .build();
+
+        if (!containsComment(comments, comment)) {
+          comments.add(comment);
         }
-        comments.add(comment);
       }
     }
 
     return comments;
   }
 
-  private Comment extractSingleComment(List<Comment> comments, String html, String author) {
-    final int startingPoint = commentStartingPoint(html, author);
-    try {
-      final String possibleComment = html.substring(startingPoint);
-      String firstCommentContentForAuthor = getFirstCommentContentForAuthor(possibleComment);
-      LocalDateTime commentDate = getCommentDate(possibleComment);
-
-      Comment firstComment = new Comment(author, firstCommentContentForAuthor, commentDate);
-      if(comments.contains(firstComment)) {
-        return searchForCorrectComment(firstCommentContentForAuthor.substring(0,20), author, html);
-      } else {
-        return firstComment;
+  private static boolean containsComment(List<Comment> c, Comment comment) {
```

Comment:
ta metoda chyba nie musi byc statyczna, plus nazwe listy `c` zamienilbym na `comments` tak dla zasady, zeby sie czepiac :)

Reply:
Jak pierwszy do wykonania takich poprawek. Tylko dalej te problemy z branchami.
```
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/feature-extractor-with-regex.
remote: error: Required status check "default" is expected.
To https://github.com/bpiatek/bbgh-backend
 ! [remote rejected] feature-extractor-with-regex -> feature-extractor-with-regex (protected branch hook declined)
error: failed to push some refs to 'https://github.com/bpiatek/bbgh-backend'

```
to chyba przez to, że branch jest protected, ale powiem szczerze że nie znam się na tym

Reply:
powinno byc ok teraz